va/x11/va_dricommon.c

```diff
@@ -71,7 +71,10 @@ do_drawable_hash(VADriverContextP ctx, XID drawable)
         dri_drawable = dri_drawable->next;
     }
 
-    dri_drawable = dri_state->createDrawable(ctx, drawable);
+    if(dri_state->createDrawable)
+    {
+        dri_drawable = dri_state->createDrawable(ctx, drawable);
+    }
     dri_drawable->x_drawable = drawable;
```

Comment:
There is no guarantee that `dri_drawable` isn't NULL here, that's what `dri_state->createDrawable()` is for I guess. So you could add another conditional to check if `dri_drawable` is null, but then you'd get to `dri_state->drawable_hash[index] = dri_drawable;`, and then you'd return a NULL dri_drawable, etc.

So I'm wondering if the problem isn't much, much deeper than a superficial check here. I'm not familiar at all with libva or how vaapi drivers work, but it seems to me that there is some missing null check in `ctx` somewhere. Some assumption about Xorg is violated when using Xwayland, and this specifically happens on iHD only, so maybe the fix really belongs there.

Reply:
driver checks nothing but call va func 'va_dri_get_drawable' to get the drawable. And func pointer 'createDrawable' is inited in va_isDRI2Connected. since iHD uses dri3 instead of dri2, then this pointer is not inited. so we saw the null pointer when call 'createDrawable'.

Reply:
Ok, that sounds like a more involved fix which requires something like https://github.com/intel/libva/pull/180 to work. I guess you're not looking to fix this by implementing that. If that's the case, why not just do
```C
if !(dri_state->createDrawable)
{
    return dri_drawable
}
```

That gives the same result as what you have right now? Basically you're just accepting that `dri_drawable` will return NULL.